### PR TITLE
Branch multiple appointments

### LINKED ISSUE #80: Add support for multiple appointments

### PR DESCRIPTION
**Pull request by jasmiinee**
Fixes #80

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ListAppointmentsCommand.java [hunk 1]
```diff
@@ -77,8 +78,14 @@ private String formatAppointments(List<Person> appointments) {
 
         return appointments.stream()
                 .map(person -> {
-                    LocalDateTime dateTime = LocalDateTime.parse(person.getSchedule().dateTime, inputFormatter);
-                    return String.format("%s: %s", person.getName(), dateTime.format(outputFormatter));
+                    Name personName = person.getName();
```
Comment (sarahteonin) [inline src/main/java/seedu/address/logic/commands/ListAppointmentsCommand.java:81]: Is this code a bit nested? Is it possible to break the nested stream operations into smaller methods?

File: src/main/java/seedu/address/logic/commands/ScheduleCommand.java [hunk 2]
```diff
@@ -120,8 +136,8 @@ private static LocalDateTime getLocalDateTime(String dateTime) {
      */
     private boolean isTimeSlotTaken(List<Person> personList, String dateTime) {
         for (Person person : personList) {
-            Schedule schedule = person.getSchedule();
-            if (schedule != null && schedule.dateTime.equals(dateTime)) {
+            Set<Schedule> schedules = person.getSchedules();
+            if (schedules.stream().map(Schedule::getDateTime).anyMatch(x -> x.equals(dateTime))) {
```
Comment (sarahteonin) [inline src/main/java/seedu/address/logic/commands/ScheduleCommand.java:140]: Could this condition be separated to be clearer?

**Review by sarahteonin — APPROVED** [review]

File: src/main/java/seedu/address/logic/parser/ScheduleCommandParser.java [hunk 3]
```diff
@@ -28,14 +27,20 @@ public class ScheduleCommandParser implements Parser<ScheduleCommand> {
...
+        Set<Schedule> scheduleList = ParserUtil.parseSchedules(argMultimap.getAllValues(PREFIX_DATE),
+            argMultimap.getAllValues(PREFIX_NOTE));
+        return new ScheduleCommand(name, scheduleList);
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
```
Comment (jiayr99) [inline src/main/java/seedu/address/logic/parser/ScheduleCommandParser.java:43]: Maybe the name could be changed to areAllPrefixesPresent to clarify that it verifies multiple prefixes?

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 4]
```diff
@@ -61,18 +64,25 @@ public PersonCard(Person person, int displayedIndex) {
...
+        // Handle multiple schedules
+        DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+        DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("MMM d yyyy, h:mm a");
+
+        Set<Schedule> schedules = person.getSchedules();
+        if (schedules.isEmpty()) {
+            schedule.setText("No scheduled appointments");
         } else {
-            note.setText("Notes: " + person.getSchedule().getNotes());
+            String formattedSchedules = schedules.stream()
```
Comment (jiayr99) [inline src/main/java/seedu/address/ui/PersonCard.java:75]: Perhaps the formatting logic could be extracted to a separate method for better code readability?

File: src/main/java/seedu/address/logic/commands/ScheduleCommand.java [hunk 5]
```diff
@@ -67,23 +81,25 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_INVALID_NAME);
         }
 
-        // Check if the schedule time is valid (on the hour and within weekday working hours)
-        if (!isOnTheHour(schedule.dateTime) || !isWithinWorkingHours(schedule.dateTime)) {
-            throw new CommandException(MESSAGE_INVALID_TIME);
-        }
+        for (Schedule schedule : scheduleSet) {
+            // Check if the schedule time is valid (on the hour and within weekday working hours)
+            if (!isOnTheHour(schedule.getDateTime()) || !isWithinWorkingHours(schedule.getDateTime())) {
```
Comment (jiayr99) [inline src/main/java/seedu/address/logic/commands/ScheduleCommand.java:86]: I like that the code clearly separates the validation checks, making it easy to understand what each condition is verifying!

**Review by jiayr99 — COMMENTED** [review]
Overall well done! :)

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 6]
```diff
@@ -121,4 +123,38 @@ public static Set<Tag> parseTags(Collection<String> tags) throws ParseException
...
+     */
+    public static Set<Schedule> parseSchedules(Collection<String> dateTimes, Collection<String> notes)
+            throws ParseException {
+        requireNonNull(dateTimes);
+        final Set<Schedule> scheduleSet = new HashSet<>();
+        Iterator<String> dateTimeIterator = dateTimes.iterator();
+        Iterator<String> noteIterator = notes.iterator();
+        while (dateTimeIterator.hasNext()) {
+            if (noteIterator.hasNext()) {
+                scheduleSet.add(parseSchedule(dateTimeIterator.next(), noteIterator.next()));
```
Comment (JumpyJay) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:153]: Could perhaps add tests that cover this line

File: src/main/java/seedu/address/logic/parser/ScheduleCommandParser.java [hunk 7]
```diff
@@ -28,14 +27,20 @@ public class ScheduleCommandParser implements Parser<ScheduleCommand> {
...
-        try {
-            String name = argMultimap.getPreamble();
-            String date = argMultimap.getValue(PREFIX_DATE).orElse("");
-            LocalDateTime dateTime = LocalDateTime.parse(date, DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
-            String note = argMultimap.getValue(PREFIX_NOTE).orElse("");
-            return new ScheduleCommand(name, new Schedule(date, note.trim()));
-        } catch (DateTimeParseException e) {
-            throw new ParseException(MESSAGE_INVALID_DATE_FORMAT);
+        if (!arePrefixesPresent(argMultimap, PREFIX_DATE)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ScheduleCommand.MESSAGE_USAGE));
```
Comment (JumpyJay) [inline src/main/java/seedu/address/logic/parser/ScheduleCommandParser.java:31]: Similarly, could add some test that covers line 31

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 8]
```diff
@@ -61,18 +64,25 @@ public PersonCard(Person person, int displayedIndex) {
...
-        }
-        if (person.getSchedule().getNotes() == null || person.getSchedule().getNotes().isEmpty()) {
-            note.setText("");
+        // Handle multiple schedules
+        DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+        DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("MMM d yyyy, h:mm a");
+
+        Set<Schedule> schedules = person.getSchedules();
+        if (schedules.isEmpty()) {
+            schedule.setText("No scheduled appointments");
```
Comment (JumpyJay) [inline src/main/java/seedu/address/ui/PersonCard.java:73]: L68-69, 71, 73 might need coverage

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 9]
```diff
@@ -61,18 +64,25 @@ public PersonCard(Person person, int displayedIndex) {
...
-            note.setText("Notes: " + person.getSchedule().getNotes());
+            String formattedSchedules = schedules.stream()
+                    .map(schedule -> {
+                        LocalDateTime dateTime = LocalDateTime.parse(schedule.getDateTime(), inputFormatter);
+                        String formattedDate = dateTime.format(outputFormatter);
+                        String noteText = schedule.getNotes();
+                        return String.format("%s [ %s ]\n", formattedDate, noteText);
+                    })
+                    .collect(Collectors.joining(""));
+            schedule.setText(formattedSchedules);
```
Comment (JumpyJay) [inline src/main/java/seedu/address/ui/PersonCard.java:83]: L75-80, 82, 83 also might require tests for coverage

**Review by JumpyJay — COMMENTED** [review]
Apart from minor code coverage, LGTM!!

**Pull request merged**